lib/vscode/src/vs/workbench/api/common/shared/webview.ts

```diff
@@ -55,9 +55,12 @@ export function asWebviewUri(
 		});
 	}
 
+	// NOTE@coder: Add the port separately because if the port is in the domain the
+	// URL will be invalid and the browser will not request it.
+	const authorityUrl = new URL(`http://${resource.authority}`);
```

Comment:
Are we guaranteed that the incoming authority is a valid URL? otherwise as Grey mentioned before, we'd have to try/catch a possible error right?

Also it does't matter since we're not using the scheme part, but you might want to consider using https here just because it looks a little nicer

Reply:
AFAIK `authority` may represent any fully qualified host e.g. `localhost:8080`, `example.com/any/path`. The  scheme [may not necessarily be http ](https://github.com/cdr/vscode/blob/main/src/vs/base/common/network.ts)

Reply:
Good call! I'll use the `resource.scheme` here instead of `http` or `https`

Reply:
We set the remote authority here: 

https://github.com/cdr/code-server/blob/cde28a0a0e29381fa5a631da092f7065bde35ccf/src/node/routes/vscode.ts#L34

So potentially it could be blank although I'm not sure how often the host header is missing.

Reply:
Personally I think letting the error throw is fine although maybe we should catch and re-throw it with a more helpful message ("host header is invalid" or something like that).

Reply:
Although VS Code does not make the authority mandatory so we should probably gracefully handle a blank authority.